aggregate.go

```diff
@@ -0,0 +1,132 @@
+package versionbundle
+
+import (
+	"reflect"
+	"sort"
+
+	"github.com/giantswarm/microerror"
+)
+
+// Aggregate merges capabilities based on dependencies version bundles within
+// the given capabilities define for their components.
+func Aggregate(capabilities []Capability) (Aggregation, error) {
+	err := ValidateCapabilities(capabilities).Validate()
+	if err != nil {
+		return Aggregation{}, microerror.Mask(err)
+	}
+
+	var newAggregration Aggregation
+
+	if len(capabilities) == 0 {
+		return newAggregration, nil
+	}
+
+	if len(capabilities) == 1 {
+		newAggregration.Capabilities = append(newAggregration.Capabilities, []Capability{capabilities[0]})
+		return newAggregration, nil
+	}
+
+	for _, c1 := range capabilities {
+		c1Copy := c1.Copy()
```

Comment:
I wanted to play it safe since we manipulate slices and slices of objects. I want to be sure all the filtering, aggregation and structure merging does not create conflicts between the objects.

Reply:
During writing tests for `Copy` I noticed that it was useless. So I removed it completely since it was not an issue as I thought beforehand. The tests still pass.